mkosi

```diff
@@ -3032,6 +3032,101 @@ class BooleanAction(argparse.Action):
         setattr(namespace, self.dest, new_value)
 
 
+class ArgumentParserMkosi(argparse.ArgumentParser):
+    """ArgumentParser with support for mkosi.defaults file(s)
+
+    This derived class adds a simple ini file parser to python's ArgumentParser features.
+    Each line of the ini file is converted to a command line argument. Example:
+    "FooBar=Hello_World"  in the ini file appends "--foo-bar Hello_World" to sys.argv.
+
+    Command line arguments starting with - or --are considered as regular arguments. Arguments
+    starting with @ are considered as files which are fed to the ini file parser implemented
+    in this class.
+    """
+
+    # Mapping of parameters supported in config files but not as command line arguments.
+    SPECIAL_MKOSI_DEFAULT_PARAMS = {
+        'QCow2': '--qcow2',
+        'OutputDirectory': '--output-dir',
+        'XZ': '--xz',
+        'NSpawnSettings': '--settings',
+        'ESPSize': '--esp-size',
+        'CheckSum': '--checksum',
+        'BMap': '--bmap',
+        'Packages': '--package',
+        'ExtraTrees': '--extra-tree',
+        'SkeletonTrees': '--skeleton-tree',
+        'BuildPackages': '--build-package',
+        'PostInstallationScript': '--postinst-script',
```

Comment:
I don't like much the different names for command line options and for configuration files. It's much easier when they match. Plural vs non plural and the case sensitive parser IMO makes it more difficult to use. Can we use this moment to fix that (possibly adding compatibility options)?

Reply:
The Python code already does what you suggest.
When ExtraTree is written to the configuration file, mkosi tries to find the ExtraTree key in this dictionary. Since this fails, ExtraTree is converted to --extra-tree.
When ExtraTree**s** is written to the configuration file, mkosi uses the transformation from the dictionary. The purpose of this dictionary is backward compatibility.

I will add another commit that removes some "s" from the man page.